packages/frontend/core/src/modules/multi-tab-sidebar/multi-tabs/tabs/chat.tsx

```diff
@@ -21,12 +21,16 @@ const EditorChatPanel = ({ editor }: SidebarTabProps) => {
     if (!editor) return;
     const pageService = editor.host.spec.getService('affine:page');
 
-    pageService.slots.docLinkClicked.on(() => {
-      (chatPanelRef.current as ChatPanel).doc = editor.doc;
-    });
-    pageService.slots.editorModeSwitch.on(() => {
-      (chatPanelRef.current as ChatPanel).host = editor.host;
-    });
+    const disposable = [
+      pageService.slots.docLinkClicked.on(() => {
+        (chatPanelRef.current as ChatPanel).doc = editor.doc;
+      }),
+      pageService.docModeService.onModeChange(() => {
+        (chatPanelRef.current as ChatPanel).host = editor.host;
```

Comment:
This editor.host may be undefined if you switch the editor mode. You can reproduce this issue with the following steps:

Open a document.
Open the chat panel on the right side.
Switch the editor mode.

Reply:
Thanks! This is a very good point I overlooked. 

The `editor.host` type is forced as `EditorHost` in blocksuite, which might be undefined. I think we should fix this in another PR.

![image](https://github.com/toeverything/AFFiNE/assets/12724894/47362bb8-4754-41f5-9b60-9ffcda273c93)
 
I have fixed this problem by detect the existence of `editor.host` and everything works fine, cause the chat panel will re-render affter mode switched and the mode$ works like behavior subject.